src/graph/unit_graph.cc

```diff
@@ -1648,7 +1649,14 @@ bool UnitGraph::Load(dmlc::Stream* fs) {
   int64_t save_format_code, formats_code;
   CHECK(fs->Read(&save_format_code)) << "Invalid format";
   CHECK(fs->Read(&formats_code)) << "Invalid format";
-  auto save_format = static_cast<SparseFormat>(save_format_code);
+  dgl_format_code_t save_formats = NONE_CODE;
+  if (save_format_code >> 32) {
+    save_formats =
+        static_cast<dgl_format_code_t>(0xffffffff & save_format_code);
```

Comment:
Put overflow handling to `SparseFormatsToCode`?

Reply:
do you mean moving the logic of `if (save_format_code >> 32)` into `SparseFormatsToCode()? That's not appropriate as `SparseFormatsToCode()` accept arguments of type `vector<SparseFormat>`.